tika-parsers/src/main/java/org/apache/tika/parser/tagRatio/TextToTagRatio.java

```diff
@@ -0,0 +1,209 @@
+            JSONObject obj = new JSONObject();
+            obj.put("avgTagRatio", (tagRatio/count));
+            obj.put("content", sb.toString());
+            
+            JSONArray array = new JSONArray();
+            array.put(metaTagsMap);
+            obj.put("meta-tags", array);
+            
+            
+            br.close();
+            
+        }catch (MalformedURLException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ParserException e) {
+            e.printStackTrace();
+            return linkTagList;
+        }
+        return linkTagList;
+    }
+    
+    /**
+     * Populate the metadata information
+     * @param line
+     * @param metaTagsMap Hash Map to store metadata as key value pairs
+     * @return
+     */
+    
+    private void populateMetaTags(String line, HashMap<String, String> metaTagsMap) {
+        
+        if(line.startsWith("<meta ")){
+            
+            Document doc = Jsoup.parse(line);
+            Element tag = doc.select("meta").first();
+            String name = tag.attr("name");
+            String content = tag.attr("content");
+            
+            metaTagsMap.put(name, content);
+        }
+    }
+    
+    /**
+     * Calculate Text to Tag Ratio for line
+     * @param line Line in the raw html
+     * @return Tag Ratio of the line
+     */
+    private double computeTextToTagRatio(String line) {
+        int tag = 0;
+        int text = 0;
+        
+        for (int i = 0; i >= 0 && i < line.length(); i++) {
+            if (line.charAt(i) == '<') {	//start tag
+                tag++;
+                i = line.indexOf('>', i);
+                if (i == -1) {
+                    break;
+                }
+            } else if (tag == 0 && line.charAt(i) == '>') {	//close tag
+                text = 0;
+                tag++;
+            } else {		//just text
+                text++;
+            }
+            
+        }
+        if (tag == 0) {
+            tag = 1;
+        }
+        if(text != 0){
+        }
+        return (double) text / (double) tag;
+    }
+    
+    
+    /**
+     * Read Xhtml data from output of Tika's AutoDetect Parser
+     * @param xhtmlOutput Xhtml output from Tika passed as input to Tag Ratio Parser
+     * @return html string
+     */
+    
+    public String readXhtmlData(String xhtmlOutput) {
+        
+        // convert String into InputStream
+        InputStream is = new ByteArrayInputStream(xhtmlOutput.getBytes());
+        
+        // read it with BufferedReader
+        BufferedReader br = new BufferedReader(new InputStreamReader(is));
+        
+        String line = null;
+        StringBuffer lines = new StringBuffer();
+        try {
+            while ((line = br.readLine()) != null) {
+                lines.append(line);
+                lines.append("\n");
+            }
+            return lines.toString();
```

Comment:
Good point @thammegowda. I was not familiar with this IOUtils package before. This will certainly improve performance.